cpp/models/abm/world.h

```diff
@@ -42,33 +45,98 @@ namespace abm
  * @brief The World of the Simulation.
  * It consists of Location%s and Person%s (Agents).
  */
-class World
+class World : public CompartmentalModel<InfectionState, Populations<AgeGroup, InfectionState>, SimulationParameters>
```

Comment:
Why does World inherit from the CompartmentalModel? There a a lot of things, that are not suited for the ABM (like flow-function, etc.). Also, parameters are saved multiple times as it seems (with the introduction of the SimulationParams).
If it is needed for the analyze_result, then there should be usage of the ParameterSets that we use in the ABM, I suppose.

Reply:
After reconsidered, I remove the CompartmentalModel from the World. However, the save_results and  analyze_result function still needs the World to have a Parameters and Populations. So I implemented them in World. Maybe it could later be removed with the new output object/function?

Reply:
As far as I can tell, populations was only needed in the analyse_result function, where you removed it. I agree with that. The parameters is a bit more difficult. In another comment, I suggested to not save parameters again, but I see the point if we want to use the save_result(s) functions. Keep in mind that we have Infection Parameters, Migration Parameters and the number of AgeGroups. So maybe we can bundle those in one parameter struct that is then publicly available in the World object? Please add your ideas @mknaranja @xsaschako.